lib/blockchain/ethereum.go

```diff
@@ -1220,6 +1256,71 @@ func (e *EthereumClient) AvgBlockTime(ctx context.Context) (time.Duration, error
 	return averageBlockTime, nil
 }
 
+// InitializeHeaderSubscription initializes either subscription-based or polling-based header processing
+func (e *EthereumClient) InitializeHeaderSubscription() error {
+	if e.Client.Client().SupportsSubscriptions() {
+		return e.subscribeToNewHeaders()
+	}
+	// Fallback to polling if subscriptions are not supported
+	e.l.Info().Str("Network", e.NetworkConfig.Name).Msg("Subscriptions not supported. Using polling for new headers.")
+	return e.startPollingHeaders()
+}
+
+// startPollingHeaders starts a polling loop to fetch new headers at regular intervals
+func (e *EthereumClient) startPollingHeaders() error {
+	pollInterval := time.Second * 5
+	ticker := time.NewTicker(pollInterval)
+	e.subscriptionWg.Add(1)
+
+	latestHeader, err := e.HeaderByNumber(context.Background(), nil) // Fetch the latest header
+	if err != nil {
+		return fmt.Errorf("failed to fetch the latest header during initialization: %w", err)
+	}
+	go func() {
+		defer e.subscriptionWg.Done()
+		// Initialize lastHeaderNumber dynamically
+		lastHeaderNumber := latestHeader.Number.Uint64() - 1
```

Comment:
why `-1`? in the first run we could could just execute `e.receiveHeader(latestHeader)` after fetching last header.
won't doing `-1` always result in the latest block being fetched 2x in every subsequent run?